barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/rom_ram_logic.cpp

```diff
@@ -201,35 +216,33 @@ void RomRamLogic_<ExecutionTrace>::process_ROM_array(CircuitBuilder* builder, co
             .record_witness = 0,
             .gate_index = 0,
         };
+        // the position of the sorted ROM gate in the execution trace depends on the witness data.
         create_sorted_ROM_gate(builder, sorted_record);
 
         builder->assign_tag(record.record_witness, read_tag);
         builder->assign_tag(sorted_record.record_witness, sorted_list_tag);
 
         // For ROM/RAM gates, the 'record' wire value (wire column 4) is a linear combination of the first 3 wire
-        // values. However...the record value uses the random challenge 'eta', generated after the first 3 wires are
-        // committed to. i.e. we can't compute the record witness here because we don't know what `eta` is! Take the
-        // gate indices of the two rom gates (original read gate + sorted gate) and store in `memory_records`. Once
-        // we
-        // generate the `eta` challenge, we'll use `memory_records` to figure out which gates need a record wire
-        // value
+        // values. However, the record value uses the random challenge 'eta', generated after the first 3 wires are
+        // committed to. i.e., we can't compute the record witness here because we don't know what `eta` is! Take the
+        // gate indices of the two rom gates (original read gate + sorted gate) and store in `memory_records`. Once we
+        // generate the `eta` challenge, we'll use `memory_records` to figure out which gates need a record wire value
         // to be computed.
-        // record (w4) = w3 * eta^3 + w2 * eta^2 + w1 * eta + read_write_flag (0 for reads, 1 for writes)
+        //
+        // `record` (w4) = w3 * eta^3 + w2 * eta^2 + w1 * eta + read_write_flag (0 for reads, 1 for writes)
         // Separate containers used to store gate indices of reads and writes. Need to differentiate because of
         // `read_write_flag` (N.B. all ROM accesses are considered reads. Writes are for RAM operations)
         builder->memory_read_records.push_back(static_cast<uint32_t>(sorted_record.gate_index));
         builder->memory_read_records.push_back(static_cast<uint32_t>(record.gate_index));
     }
-    // One of the checks we run on the sorted list, is to validate the difference between
-    // the index field across two gates is either 0 or 1.
-    // If we add a dummy gate at the end of the sorted list, where we force the first wire to
-    // equal `m + 1`, where `m` is the maximum allowed index in the sorted list,
-    // we have validated that all ROM reads are correctly constrained
+    // One of the checks we run on the sorted list is to validate the difference between the index field across two
+    // adjacent gates is either 0 or 1. To make this work with the last gate, we add a dummy gate at the end of the
```

Comment:
The second sentence is reading a little funny to me - maybe it got a bit garbled? In particular the "then all of the .." clause seems out of place